api/src/transactions.rs

```diff
@@ -406,6 +412,13 @@ mod tests {
         );
 
         let user_txn_info = expected_txns.proof.transaction_infos[1].clone();
+        let (pk, sig) = match txn.authenticator() {
+            TransactionAuthenticator::Ed25519 {
+                public_key,
+                signature,
+            } => (public_key, signature),
+            _ => panic!(),
```

Comment:
If this is unreachable for some reason, use `unreachable!`, and otherwise, this should probably have a message. Bare panic seems not great.

Reply:
These are reachable code if the code above `let txn = context.create_parent_vasp(&account);` returns a transaction with different signature type.
Thus we panic. However, it's unlikely happen.
Also these code is under test mod, thus we direct panic.
Feels like we need move test code into specific test file, as it is confusing to see a `panic` / `unwrap` in the production code file.